src/services/ExperianService.ts

```diff
@@ -0,0 +1,143 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Constants } from "../utils/Constants";
+import axios, { AxiosRequestConfig } from "axios";
+import { AppError } from "../utils/AppError";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { ExperianTokenResponse } from "../models/IExperianResponse";
+import { sleep } from "../utils/Sleep";
+import { logResponseCode } from "../utils/LogResponseCode";
+
+export class ExperianService {
+	readonly logger: Logger;
+
+	private static instance: ExperianService;
+
+    readonly experianBaseUrl: string;
+
+    readonly backoffPeriodMs: number;
+
+    readonly maxRetries: number;
+
+    constructor(logger: Logger, experianBaseUrl: string, backoffPeriodMs: number, maxRetries: number) {
+    	this.logger = logger;
+    	this.experianBaseUrl = experianBaseUrl;
+    	this.backoffPeriodMs = backoffPeriodMs;
+    	this.maxRetries = maxRetries;
+    }
+
+    static getInstance(logger: Logger, experianBaseUrl: string, backoffPeriodMs: number, maxRetries: number): ExperianService {
+    	if (!ExperianService.instance) {
+    		ExperianService.instance = new ExperianService(logger, experianBaseUrl, backoffPeriodMs, maxRetries);
+    	}
+    	return ExperianService.instance;
+    }
+
+    // eslint-disable-next-line max-lines-per-function
+    async verify(
+    	{ accountNumber, sortCode, name, uuid }: { accountNumber: string; sortCode: string; name: string; uuid: string }, token: string,
+    ): Promise<any> {
+    	const params = {
+    		header: {
+			  tenantId: uuid,
+			  requestType: "BAVConsumer-Standard",
```

Comment:
Can we move "BAVConsumer-Standard" to a variable so its understood this is a product name

Reply:
Do you mean set it in Constants? Like static readonly EXPERIAN_PRODUCT_NAME = "BAVConsumer-Standard"

Reply:
@Stelios-CB